src/main/java/net/minestom/server/instance/BlockBatchImpl.java

```diff
@@ -0,0 +1,230 @@
+package net.minestom.server.instance;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.longs.*;
+import net.kyori.adventure.nbt.CompoundBinaryTag;
+import net.minestom.server.coordinate.Vec;
+import net.minestom.server.instance.block.Block;
+import net.minestom.server.instance.block.BlockHandler;
+import net.minestom.server.instance.generator.Generator;
+import net.minestom.server.instance.palette.Palette;
+import org.jetbrains.annotations.NotNullByDefault;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static net.minestom.server.coordinate.CoordConversion.*;
+
+@NotNullByDefault
```

Comment:
should replace the not null by defaults with package info ones after rebase if there's any new packages (otherwise remove of course)